src/hooks/useColumnWidths.ts

```diff
@@ -16,6 +15,7 @@ export function useColumnWidths<R, SR>(
   setMeasuredColumnWidths: StateSetter<ReadonlyMap<string, number>>,
   onColumnResize: DataGridProps<R, SR>['onColumnResize']
 ) {
+  const [columnToAutoResize, setColumnToAutoResize] = useState<string | null>(null);
   const prevGridWidthRef = useRef(gridWidth);
```

Comment:
@amanmahajan7 https://github.com/adazzle/react-data-grid/pull/3746/files#diff-e002f181d41ce4fd6e070724347a44e4220d0332c37e5964d04cb3fc5d60d416R19
I think you better not use ref for the previous grid width. There is a bug with the initial render when:
1. table rendered with a different width from the previous
2. columns calculated + updated
3. table rendered with a different width from the previous
4. columns calculated but aren't updated -> there are no re-render -> cells will have "measurement" size (like "max-content") and will change size on content size

This issue is reflected in my video from this comment: https://github.com/adazzle/react-data-grid/issues/3723#issuecomment-2704255982

Reply:
Can you check if this issues still exists in my PR? I think the issue you are describing was related to https://github.com/adazzle/react-data-grid/pull/3724#issuecomment-2686081416